pyfar/dsp/dsp.py

```diff
@@ -51,30 +51,66 @@ def group_delay(signal, frequencies=None):
     frequencies : number array like
         Frequency or frequencies in Hz at which the group delay is calculated.
         The default is None, in which case signal.frequencies is used.
+    method : 'scipy', 'fft', optional
+        Method to calculate the group delay of a Signal. Both methods calculate
+        the group delay using the method presented in [1]_ avoiding issues
+        due to discontinuities in the unwrapped phase. Note that the scipy
+        version additionally allows to specify frequencies for which the
+        group delay is evaluated. The default is 'fft'.
 
     Returns
     -------
     group_delay : numpy array
         Frequency dependent group delay in samples. The array is flattened if
         a single channel signal was passed to the function.
+
+    References
+    ----------
+    .. [1]  https://www.dsprelated.com/showarticle/69.php
     """
 
     # check input and default values
     if not isinstance(signal, Signal):
         raise TypeError('Input data has to be of type: Signal.')
 
+    if frequencies is not None and method == 'fft':
+        raise ValueError(
+            "Specifying frequencies is not supported using the chosen method.")
```

Comment:
This would be more verbose:
```suggestion
            "Specifying frequencies is not supported for method 'fft'.")
```